packages/services/src/view-service/fees.ts

```diff
@@ -1,15 +1,27 @@
 import { AssetId } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/asset/v1/asset_pb';
 import { TransactionPlannerRequest } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/view/v1/view_pb';
+import { IndexedDbInterface } from '@penumbra-zone/types/indexed-db';
 
-// Attempts to extract a fee token from the assets used in the actions of the planner request
-// Priority in descending order
-export const extractAltFee = (request: TransactionPlannerRequest): AssetId => {
+export const extractAltFee = async (
+  request: TransactionPlannerRequest,
+  indexedDb: IndexedDbInterface,
+): Promise<AssetId> => {
   const outputAsset = request.outputs.map(o => o.value?.assetId).find(Boolean);
   if (outputAsset) return outputAsset;
 
   const swapAsset = request.swaps.map(assetIn => assetIn.value?.assetId).find(Boolean);
   if (swapAsset) return swapAsset;
 
+  const swapCommitment = request.swapClaims
+    .map(swapClaim => swapClaim.swapCommitment)
+    .find(commitment => commitment !== undefined);
```

Comment:
You can simplify this like the others to:

```diff
- .find(commitment => commitment !== undefined);
+ .find(Boolean)
```

Reply:
simplified